src/library/coap.cpp

```diff
@@ -412,6 +412,8 @@ void Coap::CancelRequests()
         mRequestsCache.Front().mRequest->GetUriPath(uri).IgnoreError();
         FinalizeTransaction(mRequestsCache.Front(), nullptr, ERROR_CANCELLED("request to {} was cancelled", uri));
     }
+
+    mRequestsCache.StopTimer();
```

Comment:
should the retransmission timer be stopped in FinalizeTransaction() when there is no ongoing request entry?

if the response arrives, should the retransmissiontimer be stopped if there is no ongoing request? better to stop retransmission timer in

Reply:
Good point. Agree that it is better to handle this inner in the `RequestsCache` class. Updated.